### PR TITLE
[SAR-3472] added JointVenturePropertyController and routes

### PR DESCRIPTION
**Pull request by LadonJackson**
Shouldn't be merged until after #489

### CODE REVIEW

File: app/uk/gov/hmrc/vatsignupfrontend/controllers/agent/partnerships/JointVenturePropertyController.scala [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      Future.successful(
+        jointVentureOrPropertyForm(isAgent = true).bindFromRequest.fold(
+          formWithErrors => BadRequest(joint_venture_or_property(
+            jointVenturePropertyForm = formWithErrors,
+            routes.JointVenturePropertyController.submit()
+          )),
+          {
+            case Yes => Redirect(routes.CheckYourAnswersPartnershipController.show())
+              .removingFromSession(SessionKeys.partnershipSautrKey)
+              .addingToSession(SessionKeys.jointVentureOrPropertyKey -> "true")
```
Comment (JenejVCap) [inline app/uk/gov/hmrc/vatsignupfrontend/controllers/agent/partnerships/JointVenturePropertyController.scala]: i believe this should also be true.toString

File: it/uk/gov/hmrc/vatsignupfrontend/controllers/agent/partnerships/JointVenturePropertyControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    }
+  }
+
+  "POST /joint-venture-or-property-partnership" when {
+
+    "form value is YES" should {
+
+      "redirect to Check Your Answers Partnership page" in {
+
+        val res = post("/client/joint-venture-or-property-partnership", Map(SessionKeys.partnershipSautrKey -> "utr"))(JointVentureOrPropertyForm.yesNo -> option_yes)
```
Comment (JenejVCap) [inline it/uk/gov/hmrc/vatsignupfrontend/controllers/agent/partnerships/JointVenturePropertyControllerISpec.scala]: maybe a valid utr could avoid some confusion, there is a testUtr that you can use

File: it/uk/gov/hmrc/vatsignupfrontend/controllers/agent/partnerships/JointVenturePropertyControllerISpec.scala [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+        res should have(
+          httpStatus(SEE_OTHER),
+          redirectUri(routes.CheckYourAnswersPartnershipController.show().url)
+        )
+
+        SessionCookieCrumbler.getSessionMap(res).get(SessionKeys.partnershipSautrKey) shouldBe None
+        SessionCookieCrumbler.getSessionMap(res).get(SessionKeys.jointVentureOrPropertyKey) shouldBe Some("true")
```
Comment (JenejVCap) [inline it/uk/gov/hmrc/vatsignupfrontend/controllers/agent/partnerships/JointVenturePropertyControllerISpec.scala]: "true" seems unnecessary, while it does the job, it also makes me want to post sad ponies gif like 

https://media.giphy.com/media/Nl6GOXV4p07te/giphy.gif

File: test/uk/gov/hmrc/vatsignupfrontend/controllers/agent/partnerships/JointVenturePropertyControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+import play.api.test.Helpers._
+import uk.gov.hmrc.vatsignupfrontend.config.featureswitch.JointVenturePropertyJourney
+import uk.gov.hmrc.vatsignupfrontend.controllers.ControllerSpec
+import uk.gov.hmrc.vatsignupfrontend.assets.MessageLookup.{AgentJointVentureOrProperty => Messages, Base => BaseMessages}
+import uk.gov.hmrc.http.NotFoundException
+import uk.gov.hmrc.vatsignupfrontend.forms.submapping.YesNoMapping._
+import uk.gov.hmrc.vatsignupfrontend.forms.JointVentureOrPropertyForm.yesNo
+import uk.gov.hmrc.vatsignupfrontend.SessionKeys
+
+
```
Comment (JenejVCap) [inline test/uk/gov/hmrc/vatsignupfrontend/controllers/agent/partnerships/JointVenturePropertyControllerSpec.scala]: can ditch some of these \n

File: test/uk/gov/hmrc/vatsignupfrontend/controllers/agent/partnerships/JointVenturePropertyControllerSpec.scala [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+        }
+
+        s"add ${SessionKeys.jointVentureOrPropertyKey} = $option_no to session" in {
+          session(result).get(SessionKeys.jointVentureOrPropertyKey) shouldBe Some(false.toString)
+        }
+      }
+
+      "An error is submitted" should {
+
+        lazy val result = TestJointVenturePropertyController.submit(testPostRequest("how?"))
```
Comment (JenejVCap) [inline test/uk/gov/hmrc/vatsignupfrontend/controllers/agent/partnerships/JointVenturePropertyControllerSpec.scala]: how indeed :P

File: app/uk/gov/hmrc/vatsignupfrontend/controllers/agent/partnerships/JointVenturePropertyController.scala [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    authorised() {
+      Future.successful(
+        jointVentureOrPropertyForm(isAgent = true).bindFromRequest.fold(
+          formWithErrors => BadRequest(joint_venture_or_property(
+            jointVenturePropertyForm = formWithErrors,
+            routes.JointVenturePropertyController.submit()
+          )),
+          {
+            case Yes => Redirect(routes.CheckYourAnswersPartnershipController.show())
+              .removingFromSession(SessionKeys.partnershipSautrKey)
```
Comment (MJCallahanPage) [inline app/uk/gov/hmrc/vatsignupfrontend/controllers/agent/partnerships/JointVenturePropertyController.scala]: Also need to remove `businessPostcodeKey`

File: app/uk/gov/hmrc/vatsignupfrontend/controllers/agent/partnerships/JointVenturePropertyController.scala [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+              .addingToSession(SessionKeys.jointVentureOrPropertyKey -> true.toString)
```
Comment (MJCallahanPage) [inline app/uk/gov/hmrc/vatsignupfrontend/controllers/agent/partnerships/JointVenturePropertyController.scala]: Change to use YesNoSessionFormatter (see my PR here #489)

File: app/uk/gov/hmrc/vatsignupfrontend/controllers/agent/partnerships/JointVenturePropertyController.scala [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+            case No => Redirect(routes.CapturePartnershipUtrController.show())
+              .addingToSession(SessionKeys.jointVentureOrPropertyKey -> false.toString)
```
Comment (MJCallahanPage) [inline app/uk/gov/hmrc/vatsignupfrontend/controllers/agent/partnerships/JointVenturePropertyController.scala]: Change to use YesNoSessionFormatter (see my PR here #489)

**Pull request merged**